src/main/kotlin/com/exactpro/th2/lwdataprovider/RequestContext.kt

```diff
@@ -18,23 +18,31 @@ package com.exactpro.th2.lwdataprovider
 
 import com.exactpro.cradle.messages.StoredMessage
 import com.exactpro.th2.common.grpc.Message
-import com.exactpro.th2.common.grpc.RawMessage
+import com.exactpro.th2.common.grpc.RawMessage as ProtoRawMessage
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.ParsedMessage
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.RawMessage
+import com.exactpro.th2.lwdataprovider.transport.toRawMessage
 import com.exactpro.th2.lwdataprovider.grpc.toRawMessage
+import com.exactpro.th2.lwdataprovider.metrics.DecodingMetrics
+import com.exactpro.th2.lwdataprovider.workers.CradleRequestId
+import com.exactpro.th2.lwdataprovider.workers.RequestId
 import java.util.concurrent.CompletableFuture
 import java.util.concurrent.TimeUnit
 import java.util.concurrent.TimeoutException
 
 class RequestedMessageDetails(
     val storedMessage: StoredMessage,
-    val group: String? = null,
+    val sessionGroup: String? = null,
     private val onResponse: ((RequestedMessageDetails) -> Unit)? = null
 ) {
-    val id: String = storedMessage.id.toString()
-    val rawMessage: RawMessage = storedMessage.toRawMessage()
+    val requestId: RequestId = CradleRequestId(storedMessage.id)
+    val protoRawMessage: Lazy<ProtoRawMessage> = lazy(LazyThreadSafetyMode.NONE, storedMessage::toRawMessage)
+    val transportRawMessage: Lazy<RawMessage> = lazy(LazyThreadSafetyMode.NONE, storedMessage::toRawMessage)
```

Comment:
Usually, Lazy is used as property delegate
```
```suggestion
    val protoRawMessage: ProtoRawMessage by lazy(LazyThreadSafetyMode.NONE, storedMessage::toRawMessage)
    val transportRawMessage: RawMessage by lazy(LazyThreadSafetyMode.NONE, storedMessage::toRawMessage)
```

Maybe we should remove that conversion from here and leave only storedMessage here. The conversion to the corresponding format will be done in a place where this format is required